modules/createBrowserHistory.js

```diff
@@ -165,6 +165,11 @@ const createBrowserHistory = (props = {}) => {
     const action = "PUSH"
     const location = createLocation(path, state, createKey(), history.location)
 
+    if (createPath(location) === createPath(history.location)) {
```

Comment:
You're only comparing the pathname in this case. What about the search/querystring? Since you're doing this in multiple histories, you should export a common function, like [`locationsAreEqual`](https://github.com/ReactTraining/history/blob/master/modules/LocationUtils.js#L71-L76).

Reply:
Great tips, thanks.

There's already a `locationsAreEqual` method, which is great. But it also checks for `key`, something we don't want. Should I add an override to the method?

That would result in the following signature:

```js
locationsAreEqual = (a, b, ignoreKey) => { ... }
```

Sounds reasonable if you ask me. On the other hand, invoking the function gets a bit vague that way: 

```js
locationsAreEqual(location, oldLocation, true)
```

I could also create a new method `locationsExceptKeyAreEqual` or `locationEqualsDestination`. What do you think?

Reply:
Hmm, I'm not sure why key is in there at all. If the key is equal, the locations are equal. That's the point of the key value being in there. So, that function would appear to be fundamentally flawed. There's only one internal usage: 

https://github.com/ReactTraining/history/blob/dca21eaa9230ae187c72ef657f70aac97cb172aa/modules/createHashHistory.js#L110

Reply:
Oh, I see why. Hash histories don't use key. I haven't been following those specifics too carefully, since I don't use those myself.

I would add a `locationsAreEquivalent` function in this case that checks `pathname` and `search` are equal.

Reply:
Thanks for thinking along!

Looks like the key prop could be removed from the compare function, right? As hash history doesn't use the key prop anyhow. Then it would be usable in all places without adding an extra method.